Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Specify base image for building service binary
+FROM golang:1.20
+
+# Install go debugger for easier debug life
+RUN go install github.com/go-delve/delve/cmd/dlv@latest
+
+# create and set default directory for service  files
+RUN mkdir /app
+WORKDIR /app
+
+# optimize build time by installing dependencies
+# before building so if source code changed but not
+# the list of dependencies they don't have to be re-downloaded
+COPY go.mod go.sum ./
+
+# download service golang dependnecies source code
+RUN go mod tidy
+
+# copy over local sources used to build service
+COPY *.go ./
+
+# build service from latest sources
+# with all compilier optimizations off to support debugging
+RUN go install  -gcflags=all="-N -l"
```

Comment:
when it's time to deploy, will create production dockerfile that complies with optimizations & doesn't install any development only dependencies

Reply:
split out production and local dockerfiles